### DISCUSSION TITLE
[2023-01-08] Team GUI Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8705] [Weekly Demo Day]
### Summary: what left to enjoy new GUI?
* #8683 - an AST changes API required to fix some issues in setting placeholders' values and finish other tasks.
* #8494 - blocked by above.
* #8663 - otherwise we won't display placeholders for constructors' arguments.
* #8686 - in review process.

**Comment by farmaazon** [comment]
* #8687 - we finally have + button in array placeholders!

[Screencast from 2024-01-05 12-18-55.webm](https://github.com/enso-org/enso/assets/3919101/f083512f-f698-42d1-b43a-4e50546b958a)

**Comment by vitvakatu** [comment]
Not much progress since last demo:
- Finishing https://github.com/enso-org/enso/pull/8686 (on review). Proper nodes positioning requires fixes in AST editing API.
- Starting implementation of e2e tests for collapsing nodes and entering nodes. It requires some changes in our mocks.

**Comment by Frizi** [comment]
- Node reordering still in progress, currently blocked by AST edit bugs, waiting on Kaz to deliver a fix https://github.com/enso-org/enso/issues/8494
- Working on unified icons setup in Figma. Currently merged all icons from the app and in design, and unified their definition to always fit 16x16 and have consistent colors. The goal is to have a one-click export option for whole icon sheet. https://github.com/enso-org/enso/issues/8387

https://github.com/enso-org/enso/assets/919491/c4e18993-1cb8-4960-8367-f8d9d7010531

**Comment by hubertp** [comment]
Relevant backend changes:
- Using simple data I/O streams instead of Jackson in ModuleCache #8693 
- Cats are not welcome #8696 
- Remove org.bouncycastle dependency #8664 
- Test benchmark theory #8668 
.. and possibly one more (optimizing AliasAnalysis pass #8701)

So far 13% startup improvement between 2nd and 8th of January.
